ft/template.robot

```diff
@@ -0,0 +1,214 @@
+    Check Response Status         ${res}                404
+    Check Error Code              ${res}                ${NEU_ERR_PLUGIN_NOT_FOUND}
+
+    [Teardown]                    Del Template          ${g_template}
+
+
+Create a template with an existing plugin, it should success.
+    ${res} =                      Add Template          ${g_template}                 ${g_plugin}           ${g_empty}
+    Check Response Status         ${res}                200
+    Check Error Code              ${res}                ${NEU_ERR_SUCCESS}
+
+    [Teardown]                    Del Template          ${g_template}
+
+
+Create a template with invalid group interval, it should fail.
+    ${group} =                    Create Dictionary     name=group1                   interval=10           tags=${g_tags}
+    ${groups} =                   Create List           ${group}
+
+    ${res} =                      Add Template          ${g_template}                 ${g_plugin}           ${groups}
+    Check Response Status         ${res}                400
+    Check Error Code              ${res}                ${NEU_ERR_GROUP_PARAMETER_INVALID}
+
+    [Teardown]                    Del Template          ${g_template}
+
+
+Create a template with too long group name, it should fail.
+    ${name} =                     Evaluate              "a"*200
+    ${group} =                    Create Dictionary     name=${name}                  interval=${2000}      tags=${g_tags}
+    ${groups} =                   Create List           ${group}
+
+    ${res} =                      Add Template          ${g_template}                 ${g_plugin}           ${groups}
+    Check Response Status         ${res}                400
+    Check Error Code              ${res}                ${NEU_ERR_GROUP_NAME_TOO_LONG}
+
+    [Teardown]                    Del Template          ${g_template}
+
+
+Create a template with occupied name, it should fail.
+    ${res} =                      Add Template          ${g_template}                 ${g_plugin}           ${g_empty}
+    Check Response Status         ${res}                200
+    Check Error Code              ${res}                ${NEU_ERR_SUCCESS}
+
+    ${res} =                      Add Template          ${g_template}                 ${g_plugin}           ${g_empty}
+    Check Response Status         ${res}                409
+    Check Error Code              ${res}                ${NEU_ERR_TEMPLATE_EXIST}
+
+    [Teardown]                    Del Template          ${g_template}
+
+
+Create a template with singleton plugin, it should fail.
+    ${res} =                      Add Template          ${g_template}                 Monitor               ${g_empty}
+    Check Response Status         ${res}                400
+    Check Error Code              ${res}                ${NEU_ERR_PLUGIN_NOT_SUPPORT_TEMPLATE}
+
+    [Teardown]                    Del Template          ${g_template}
+
```

Comment:
1.  add a case to create a template with some tags, it should success.
2. add a case to create a template with a tag which using error address, it should fail.
3. add a case to create a template with a tag which using error type, it should fail.